packages/kbn-scout-reporting/src/helpers/elasticsearch.ts

```diff
@@ -16,20 +16,25 @@ import { createFailError } from '@kbn/dev-cli-errors';
  *
  * @param options Elasticsearch client options
  * @param log Logger instance
+ * @param cli Set to `true` when invoked from a CLI context
  * @throws FailError if cluster information cannot be read from the target Elasticsearch instance
  */
 export async function getValidatedESClient(
   options: ESClientOptions,
-  log: ToolingLog
+  log?: ToolingLog,
+  cli: boolean = false
```

Comment:
I'm always a bit torn with plain boolean parameters, particularly when their meaning is not obvious from the function name.

On the one hand, it's harder to read in the current form, e.g. when seeing
```
const es = await getValidatedESClient(eSClientOptions, log, true);
```
in the code, I have no idea what the third parameter is about until I inspect the function.

On the other hand, wrapping the additional parameters in an object bloats the function definition. But it makes the calling side more readable:
```
const es = await getValidatedESClient(eSClientOptions, {log, cli: true});
```

I think in most cases, I prefer the second option, but that's probably a matter of taste.
What do you think?

Reply:
Yep. I agree about readability. 

Resolved in https://github.com/elastic/kibana/pull/206431/commits/77f3fc841aac2cfb459f0f52fa8560cb87014676